src/Databases/DatabaseReplicated.cpp

```diff
@@ -1463,8 +1465,11 @@ void DatabaseReplicated::recoverLostReplica(const ZooKeeperPtr & current_zookeep
                 }
 
                 auto query_ast = parseQueryFromMetadataInZooKeeper(table_name, create_query_string);
-
                 auto create_query_context = make_query_context();
+
+                NormalizeSelectWithUnionQueryVisitor::Data data{create_query_context->getSettingsRef()[Setting::union_default_mode]};
```

Comment:
nit: this probably belongs in `InterpreterCreateQuery`

Reply:
Interestingly, DatabaseBackup and DatabaseOrdinary have the same code. Maybe indeed it is worth to move it to the InterpreterCreateQuery instead.